### PR TITLE
Adjusted and improved deployment pricing

### PR DESCRIPTION
**Pull request by miloszm**
Adjusted and improved deployment pricing

Deployment pricing is now based on charge rather than gas usage, which allows for higher gas prices to be used.
Deployment charge per bytecode byte configuration item and default are now expressed as (economic) value rather than gas. This enables putting higher-priced deployment transactions into blocks without exceeding the block gas limit.

The goal is to have a reasonable contract deployment pricing, so that deployment of a small contract, say 10kB, costs around 2 DUSK, and deployment of a larger contract, say, 500kB, costs around 100 DUSK. Should we want to fine tune these values, it is enough to change the configuration value named `charge_per_deploy_byte` (current value of this setting is 200000).

Implements #2207

### CODE REVIEW

**Review by fed-franz — COMMENTED** [review]
I might be a bit unfamiliar with this part, so I'm just gonna leave comments.

I'm a bit confused by the `bytecode_charge` function. If everybody is confident on it, I'll trust you, but maybe add a line or two to explain the logics of the function, especially the `gas_charge` calculation.

Also, I would rename the `charge` variables to `byte_charge`, in contrast to `gas_charge`

File: rusk/src/lib/node/rusk.rs [hunk 1]
```diff
@@ -532,13 +533,17 @@ fn accept(
...
-) -> u64 {
-    bytecode.bytes.len() as u64
-        * gas_per_deploy_byte.unwrap_or(DEFAULT_GAS_PER_DEPLOY_BYTE)
+    charge_per_deploy_byte: &Option<u64>,
+    gas_price: u64,
+) -> (u64, u64) {
+    let charge = bytecode.bytes.len() as u64
+        * charge_per_deploy_byte.unwrap_or(DEFAULT_CHARGE_PER_DEPLOY_BYTE);
+    assert_ne!(gas_price, 0);
+    let gas_charge = max(charge / gas_price, 1);
```
[the remark this answers is not in the document]
Comment (miloszm) [reply rusk/src/lib/node/rusk.rs:545]: Currently, default charge is 200k LUX per byte. So to bring deployment gas usage to 1, the price would need to be in (even say for a small 10k contract) a BLN LUX range , i.e. several DUSK per gas point, that is practically impossible.

File: rusk/default.config.toml [hunk 2]
```diff
@@ -13,7 +13,7 @@
 #consensus_keys_path = '/home/user/.dusk/rusk/consensus.keys'
 #generation_timeout = '3s'
 # Note: changing the gas per deploy byte parameter is equivalent to forking the chain.
```
[the remark this answers is not in the document]
Comment (miloszm) [reply rusk/default.config.toml:15]: good catch

**Comment by miloszm** [comment]
We've decided for an alternative implementation

**Pull request closed without merging**